backend/api/admin.py

```diff
@@ -29,16 +29,16 @@
 #       None
 def accept_signup_request(ModelAdmin, request, queryset):
     for req in queryset:
-        a = Author(username=req.username, password=make_password(req.password), host = req.host, git_url=req.git_url)
-        a.url = f'{settings.HOST_URL}author/{a.id}'
+        a = Author(displayName=req.displayName ,username=req.username, password=make_password(req.password), host = req.host, git_url=req.git_url)
+        a.url = f'{req.host}author/{a.id}'
         a.save()
     queryset.delete()
 
 accept_signup_request.short_description = "allow them to be on server"
 
```

Comment:
Is it possible to move the signup_request function into `view` and `services`?

I dont like to have these work in `admin.py`

Reply:
Depend on what's your thinking, because this technically is not a regular view, it works based on some mechanism of admin. If you think admin.py can be in /view, I would say its probably better to move function into /services.

Reply:
In that case, we might need to have a `admin` folder to store the class

Reply:
if this is a one time thing how reasonable is it to make a new file?

Reply:
It will have couple methods and classes related to admin. 

And admin.py should not have those classes and functions inside. I would recommend to have a clear structure.